### PR TITLE
grandpa-rpc don't share subscription manager, only executor

### PR DESCRIPTION
**Pull request by octol**
Instead of having `sc-finality-grandpa-rpc` share subscription manager with the client, only share executors.
Based on the idea that less sharing is better.

Part of: #6066 
polkadot companion: https://github.com/paritytech/polkadot/pull/1687

### CODE REVIEW

**Review by tomusdrw — COMMENTED** [review]
lgtm!

File: bin/node/cli/src/service.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ pub fn new_partial(config: &Configuration) -> Result<sc_service::PartialComponen
 	(
 		impl Fn(
 			node_rpc::DenyUnsafe,
-			jsonrpc_pubsub::manager::SubscriptionManager
+			sc_rpc::SubscriptionTaskExecutor
```
Comment (andresilva) [inline bin/node/cli/src/service.rs:54]:
```suggestion
			sc_rpc::SubscriptionTaskExecutor,
```

File: client/service/src/builder.rs [hunk 2]
```diff
@@ -73,17 +73,23 @@ pub trait RpcExtensionBuilder {
 
 	/// Returns an instance of the RPC extension for a particular `DenyUnsafe`
 	/// value, e.g. the RPC extension might not expose some unsafe methods.
-	fn build(&self, deny: sc_rpc::DenyUnsafe, subscriptions: SubscriptionManager) -> Self::Output;
+	fn build(&self,
+		deny: sc_rpc::DenyUnsafe,
+		subscription_executor: sc_rpc::SubscriptionTaskExecutor
```
Comment (andresilva) [inline client/service/src/builder.rs:78]:
```suggestion
	fn build(
		&self,
		deny: sc_rpc::DenyUnsafe,
		subscription_executor: sc_rpc::SubscriptionTaskExecutor,
```

File: client/service/src/builder.rs [hunk 3]
```diff
@@ -73,17 +73,23 @@ pub trait RpcExtensionBuilder {
...
+	F: Fn(sc_rpc::DenyUnsafe, sc_rpc::SubscriptionTaskExecutor) -> R,
 	R: sc_rpc::RpcExtension<sc_rpc::Metadata>,
 {
 	type Output = R;
 
-	fn build(&self, deny: sc_rpc::DenyUnsafe, subscriptions: SubscriptionManager) -> Self::Output {
-		(*self)(deny, subscriptions)
+	fn build(&self,
+		deny: sc_rpc::DenyUnsafe,
+		subscription_executor: sc_rpc::SubscriptionTaskExecutor
```
Comment (andresilva) [inline client/service/src/builder.rs:90]:
```suggestion
	fn build(
		&self,
		deny: sc_rpc::DenyUnsafe,
		subscription_executor: sc_rpc::SubscriptionTaskExecutor,
```

File: client/service/src/builder.rs [hunk 4]
```diff
@@ -97,7 +103,10 @@ impl<R> RpcExtensionBuilder for NoopRpcExtensionBuilder<R> where
 {
 	type Output = R;
 
-	fn build(&self, _deny: sc_rpc::DenyUnsafe, _subscriptions: SubscriptionManager) -> Self::Output {
+	fn build(&self,
+		_deny: sc_rpc::DenyUnsafe,
+		_subscription_executor: sc_rpc::SubscriptionTaskExecutor
```
Comment (andresilva) [inline client/service/src/builder.rs:108]:
```suggestion
	fn build(&self,
		_deny: sc_rpc::DenyUnsafe,
		_subscription_executor: sc_rpc::SubscriptionTaskExecutor
```
```suggestion
	fn build(
		&self,
		_deny: sc_rpc::DenyUnsafe,
		_subscription_executor: sc_rpc::SubscriptionTaskExecutor,
```

**Review by andresilva — APPROVED** [review]
lgtm

File: client/service/src/builder.rs [hunk 5]
```diff
@@ -97,7 +103,10 @@ impl<R> RpcExtensionBuilder for NoopRpcExtensionBuilder<R> where
...
+	) -> Self::Output {
```
Comment (andresilva) [inline client/service/src/builder.rs:109]:
```suggestion
	fn build(
		&self,
		_deny: sc_rpc::DenyUnsafe,
		_subscription_executor: sc_rpc::SubscriptionTaskExecutor
	) -> Self::Output {
```

**Comment by andresilva** [comment]
bot merge

**Comment by andresilva** [comment]
bot merge

**Pull request merged**